blog/_posts/2022-01-25-helm-improvements.md

```diff
@@ -0,0 +1,154 @@
+
+- You're iterating on a server that has so much YAML that you need Helm to get your arms around it.
+
+We've made some big improvements to our Helm support recently with the [`helm_resource`
+extension](https://github.com/tilt-dev/tilt-extensions/tree/master/helm_resource).
+
+Let's dig into how you'd use it and why.
+
+## Why Helm?
+
+Helm is a packaging standard for Kubernetes apps.
+
+In its simplest form, a Helm chart is simply a collection of Kubernetes objects
+described in YAML, with templating support to adapt those objects to different
+environments. Your objects might look slightly different in prod vs in dev.
+
+But over the past few years, charts can do more and more at runtime, including:
+
+- Installing different objects depending on which Kubernetes version you're using.
+
+- Controlling the order that objects are installed in.
+
+- Running checks at different stages of the install process.
+
+That's made it the tool-of-choice for installing a package of servers.
+
+## Installing off-the-shelf servers
+
+Many off-the-shelf tools have Helm charts. The chart is an easy way to install a
+collection of objects into your Kubernetes cluster. You don't have to worry
+about what objects are in there or what order to install them in.
+
+Here's an example that deploys the [Bitnami `mysql`
+chart](https://artifacthub.io/packages/helm/bitnami/mysql):
+
+```python
+load('ext://helm_resource', 'helm_resource', 'helm_repo')
+helm_repo('bitnami', 'https://charts.bitnami.com/bitnami')
+helm_resource('mysql', 'bitnami/mysql')
+```
+
+Visit [Artifact Hub](https://artifacthub.io/) to find Helm charts for
+many off-the-shelf tools. They each list the repo URL and chart name to
+use with `helm_resource`.
+
+## Iterating on the Image
+
+Many Helm charts let you inject your own image. That's why the `helm_resource`
+extension lets you list a dependency on an image, then inject the image into the
+chart automatically when it changes.
+
+You need to add two arguments to your `helm_resource` declaration:
+
+- An `image_deps` that lists the images you depend on.
+
+- An `image_keys` that tells Tilt the names of the template keys for those images.
+
+A widespread convention in the Helm ecosystem is to describe image with two
+different keys: a `repository` key (e.g., `my-org/my-image-name` and a `tag` key
+(e.g., `v2`). So `helm_resource` supports a special syntax for describing this
+as a `(respository_key, tag_key)` tuple.
+
+Here's an example from the `helm_resource` tests:
+
+```python
+load('ext://helm_resource', 'helm_resource')
+
+docker_build(
+  'helloworld-image',
+  './src',
+  dockerfile='./Dockerfile')
+
+# The ./helloworld helm chart is the default
+# chart generated by 'helm create'
+helm_resource(
+  'helloworld',
+  './helloworld',
+  deps=['./helloworld'],
+  image_deps=['helloworld-image'],
+  image_keys=[('image.repository', 'image.tag')])
```

Comment:
Are these set keywords? If they can be customized, I would make it extra clear, e.g.
```suggestion
  image_keys=[('helloworld-image.repository', 'helloworld-image.tag')])
```

Reply:
they're not keywords exactly, but it's a pretty strong convention to call them image.repository and image.tag. e.g.:
https://github.com/bitnami/charts/blob/master/bitnami/mysql/values.yaml#L70